### PR TITLE
[Story] CRUD for users

### PR DESCRIPTION
**Pull request by joukevandermaas**
AB#53833

### CODE REVIEW

File: Library/Models/LocalUser.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+{
+    /// <summary>
+    /// Model representing the settable values in on a local user.
+    /// </summary>
+    public class LocalUserValues
+    {
+        /// <summary>
+        /// Name of user
+        /// </summary>
+        [JsonProperty]
```
Comment (bjorncoltof) [inline Library/Models/LocalUser.cs]: Do we also need these on all the public properties?

Comment (joukevandermaas) [reply Library/Models/LocalUser.cs]: I guess we probably don't? I will try remove them and see what happens.

File: Library/Models/LocalUser.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using Newtonsoft.Json;
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Model representing the settable values in on a local user.
```
Comment (marcoNipo) [inline Library/Models/LocalUser.cs]: "_in_ on a local user" :thinking:

File: Library/Models/LocalUser.cs [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    /// <summary>
+    /// Model representing a local user.
+    /// </summary>
+    public class LocalUser : LocalUserValues
+    {
+        /// <summary>
+        /// The Id of the local user.
+        /// </summary>
+        [JsonProperty]
```
Comment (marcoNipo) [inline Library/Models/LocalUser.cs]: Do you need this attribute?

File: Library/Services/INfieldLocalUserService.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+{
+    /// <summary>
+    /// Service used to list, update, create and delete local users.
+    /// </summary>
+    public interface INfieldLocalUserService
+    {
+        /// <summary>
+        /// Retrieve a specific local user
+        /// </summary>
+        /// <param name="identityId"></param>
```
Comment (marcoNipo) [inline Library/Services/INfieldLocalUserService.cs]: If the comment is empty, you can delete the param node

File: Library/Services/Implementation/NfieldLocalUserService.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        public async Task<LocalUser> CreateAsync(NewLocalUser model)
+        {
+            if (model == null)
+                throw new ArgumentNullException(nameof(model));
+
+            var uri = new Uri(ConnectionClient.NfieldServerUri, "LocalUsers");
+
+            using (var response = await ConnectionClient.Client.PostAsJsonAsync(uri, model))
+            {
+                var result = await DeserializeJsonAsync<LocalUser>(response);
```
Comment (marcoNipo) [inline Library/Services/Implementation/NfieldLocalUserService.cs]: you don't really need the variable

File: Library/Services/Implementation/NfieldLocalUserService.cs [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        public async Task<LocalUser> UpdateAsync(string identityId, LocalUserValues model)
+        {
+            if (model == null)
+                throw new ArgumentNullException(nameof(model));
+
+            var uri = new Uri(ConnectionClient.NfieldServerUri, $"LocalUsers/{identityId}");
+
+            using (var response = await ConnectionClient.Client.PatchAsJsonAsync(uri, model))
+            {
+                var result = await DeserializeJsonAsync<LocalUser>(response);
```
Comment (marcoNipo) [inline Library/Services/Implementation/NfieldLocalUserService.cs]: same here

File: Library/Services/Implementation/NfieldLocalUserService.cs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            }
+        }
+
+        public async Task DeleteAsync(string identityId)
+        {
+            var uri = new Uri(ConnectionClient.NfieldServerUri, $"LocalUsers/{identityId}");
+
+            // note: we need to dispose the response even when we don't use it
+            using (await ConnectionClient.Client.DeleteAsync(uri))
+            {
```
Comment (marcoNipo) [inline Library/Services/Implementation/NfieldLocalUserService.cs]: Is it possible to do it without the brackets {}? Maybe with a simple `;` after the using?

Comment (bjorncoltof) [reply Library/Services/Implementation/NfieldLocalUserService.cs]: .Dispose()?

**Pull request merged**